plugin-server/src/main/ingestion-queues/batch-processing/each-batch-webhooks.ts

```diff
@@ -0,0 +1,180 @@
+import * as Sentry from '@sentry/node'
+import { StatsD } from 'hot-shots'
+import { EachBatchPayload, KafkaMessage } from 'kafkajs'
+import { ActionMatcher } from 'worker/ingestion/action-matcher'
+
+import { PostIngestionEvent, RawClickHouseEvent } from '../../../types'
+import { DependencyUnavailableError } from '../../../utils/db/error'
+import { convertToIngestionEvent, convertToProcessedPluginEvent } from '../../../utils/event'
+import { status } from '../../../utils/status'
+import { processWebhooksStep } from '../../../worker/ingestion/event-pipeline/runAsyncHandlersStep'
+import { silentFailuresAsyncHandlers } from '../../../worker/ingestion/event-pipeline/runner'
+import { HookCommander } from '../../../worker/ingestion/hooks'
+import { runInstrumentedFunction } from '../../utils'
+import { latestOffsetTimestampGauge } from '../metrics'
+
+// Must require as `tsc` strips unused `import` statements and just requiring this seems to init some globals
+require('@sentry/tracing')
+
+function groupIntoBatches(
+    array: KafkaMessage[],
+    batchSize: number,
+    actionMatcher: ActionMatcher
+): { eventBatch: RawClickHouseEvent[]; lastOffset: string; lastTimestamp: string }[] {
+    // Most events will not trigger a webhook call, so we want to filter them out as soon as possible
+    // to achieve the highest effective concurrency when executing the actual HTTP calls.
+    // actionMatcher holds an in-memory set of all teams with enabled webhooks, that we use to
+    // drop events based on that signal. To use it we must parse the message, as there aren't that many
+    // webhooks, we can keep batches of the parsed messages in memory with the offsets of the last message
+    const result: { eventBatch: RawClickHouseEvent[]; lastOffset: string; lastTimestamp: string }[] = []
+    let currentBatch: RawClickHouseEvent[] = []
+    let currentCount = 0
+    array.forEach((message, index) => {
+        const clickHouseEvent = JSON.parse(message.value!.toString()) as RawClickHouseEvent
+        if (actionMatcher.hasWebhooks(clickHouseEvent.team_id)) {
+            currentBatch.push(clickHouseEvent)
+            currentCount++
+        }
```

Comment:
Let's add a else, call
```
                eventDroppedCounter
                    .labels({
                        event_type: 'analytics-webhook',
                        drop_cause: 'no_matching_action',
                    })
                    .inc()
```
so that we can rule out a regression there if webhooks are not sent anymore.